app/services/release_service.rb

```diff
@@ -2,30 +2,44 @@ class ReleaseService
   # @param [Purl] purl
   # @param [Hash<String, Array<String>>] where to release to. Note that even when this is empty, we release to the "always_send_true_targets"
   def self.release(purl, actions)
+    new(purl).release(actions)
+  end
+
+  attr_reader :purl
+
+  delegate :druid, to: :purl
+
+  def initialize(purl)
+    @purl = purl
+  end
+
+  def release(actions)
     # add the release tags, and reuse tags if already associated with this PURL
     purl.refresh_release_tags(actions)
-    write_meta_json(purl)
+    write_meta_json
     purl.save!
     purl.produce_indexer_log_message
   end
 
-  def self.write_meta_json(purl)
-    if VersionedFilesService.versioned_files?(druid: purl.druid)
-      VersionedFilesService::Paths.new(druid: purl.druid).meta_json_path.write(meta_json(purl))
-      write_purl_meta_json(purl) if Settings.features.legacy_purl
+  def write_meta_json
+    if VersionedFilesService.versioned_files?(druid:)
+      VersionedFilesService::Paths.new(druid:).meta_json_path.write(meta_json)
+      write_purl_meta_json if Settings.features.legacy_purl
     else
-      write_purl_meta_json(purl)
+      write_purl_meta_json
     end
   end
 
-  def self.write_purl_meta_json(purl)
+  private
+
+  def write_purl_meta_json
     file_path = File.join(purl.purl_druid_path, 'meta.json')
-    File.write(file_path, meta_json(purl))
+    File.write(file_path, meta_json)
   end
-  private_class_method :write_purl_meta_json
 
-  def self.meta_json(purl)
+  def meta_json
     {
+      '$schemaVersion': 1,
```

Comment:
Being able to make this change without changing a test seems to indicate missing tests.

Reply:
That's one of the reasons I thought this class would benefit from a little refactoring. `private_class_method` seemed like a code-smell.